pkg/pipelinerunmetrics/metrics.go

```diff
@@ -256,40 +280,77 @@ func (r *Recorder) DurationAndCount(pr *v1beta1.PipelineRun, beforeCondition *ap
 	return nil
 }
 
-// RunningPipelineRuns logs the number of PipelineRuns running right now
+// RunningPipelineRuns logs the number of PipelineRuns running right now,
+// along with whether the number of PipelineRuns throttled by k8s scheduling.
 // returns an error if its failed to log the metrics
-func (r *Recorder) RunningPipelineRuns(lister listers.PipelineRunLister) error {
+func (r *Recorder) RunningPipelineRuns(prLister listers.PipelineRunLister, trLister listers.TaskRunLister) error {
 	r.mutex.Lock()
 	defer r.mutex.Unlock()
 
 	if !r.initialized {
 		return errors.New("ignoring the metrics recording, failed to initialize the metrics recorder")
 	}
 
-	prs, err := lister.List(labels.Everything())
+	prs, err := prLister.List(labels.Everything())
 	if err != nil {
 		return fmt.Errorf("failed to list pipelineruns while generating metrics : %w", err)
 	}
 
 	var runningPRs int
+	var prsThrottledByQuota int
+	var prsThrottledByNode int
 	for _, pr := range prs {
 		if !pr.IsDone() {
 			runningPRs++
 		}
+		if pr.IsDone() {
+			continue
+		}
+		for _, cr := range pr.Status.ChildReferences {
+			switch cr.Kind {
+			case pipeline.TaskRunControllerName:
+				trName := cr.Name
+				tr, trerr := trLister.TaskRuns(pr.Namespace).Get(trName)
+				if trerr == nil {
+					if tr.IsDone() || tr.IsCancelled() {
+						continue
+					}
+					succeedCondition := tr.Status.GetCondition(apis.ConditionSucceeded)
+					exitForLoop := false
+					if succeedCondition != nil && succeedCondition.Status == corev1.ConditionUnknown {
+						switch succeedCondition.Reason {
+						case pod.ReasonExceededResourceQuota:
+							prsThrottledByQuota++
+							// only count once per pipelinerun
+							exitForLoop = true
```

Comment:
We can use break instead of `exitForLoop`?

Reply:
My recollection was that break only exited one for loop, and not both

The intent was to avoid bumping the counter for a pipelinerun multiple times because multiple taskruns are throttled.

Hence I want to leave the `for _, pr :=range prs` for loop as well.

Given that context, are you OK then with leaving this as is @khrm ?  Or do you think I'm missing something.

Reply:
wait if what I 'm saying is the case, I'm not breaking out of the last loop

yeah let me revisit @khrm .... either I break out of the last loop as well, or I just break here.

I'll update after I have fully revisited - thanks

Reply:
OK @khrm - while I did not add a break exactly here as you mentioned, you flagging this made me realize what I had previously pushed was not quite what I intended, which is:

1) if a pipelinerun is both throttled on resource quota and node constraints with one of its underlying taskruns, bump the count once for resource quota and node constraint
2) if a pipelinerun has multiple taskruns throttled by either resource quota or node constraints, don't bump the counts for each taskruns, but just once for the pipelinerun

aside from the code changes (2 booleans vs. one essentially for tracking this), I augmented the unit tests to create more than one taskrun per pipelinerun

PTAL

Reply:
Yes, the current logic looks good to me.